contracts/src/deployers/aave/AaveHyperdriveDeployerCoordinator.sol

```diff
@@ -0,0 +1,206 @@
+    /// @param _options The options that configure how the initialization is
+    ///        settled.
+    /// @return value The value that should be sent in the initialize transaction.
+    function _prepareInitialize(
+        IHyperdrive _hyperdrive,
+        address _lp,
+        uint256 _contribution,
+        IHyperdrive.Options memory _options
+    ) internal override returns (uint256) {
+        // If base is the deposit asset, the initialization will be paid in the
+        // base token.
+        address token;
+        address baseToken = _hyperdrive.baseToken();
+        if (_options.asBase) {
+            token = baseToken;
+        }
+        // Otherwise, the initialization will be paid in vault shares.
+        else {
+            // The token is the vault shares token.
+            token = _hyperdrive.vaultSharesToken();
+
+            // AToken transfers are in base, so we need to convert from
+            // shares to base.
+            _contribution = _convertToBase(baseToken, token, _contribution);
+        }
+
+        // Take custody of the contribution and approve Hyperdrive to pull the
+        // tokens.
+        ERC20(token).safeTransferFrom(_lp, address(this), _contribution);
+        ERC20(token).forceApprove(address(_hyperdrive), _contribution);
+
+        // This yield source isn't payable, so we should always send 0 value.
+        return 0;
+    }
+
+    /// @dev Prevents the contract from receiving ether.
+    function _checkMessageValue() internal view override {
+        if (msg.value != 0) {
+            revert IHyperdriveDeployerCoordinator.NotPayable();
+        }
+    }
+
+    /// @notice Checks the pool configuration to ensure that it is valid.
+    /// @param _deployConfig The deploy configuration of the Hyperdrive pool.
+    function _checkPoolConfig(
+        IHyperdrive.PoolDeployConfig memory _deployConfig
+    ) internal view override {
+        // Perform the default checks.
+        super._checkPoolConfig(_deployConfig);
+
+        // Ensure that the vault shares token address is non-zero.
+        if (address(_deployConfig.vaultSharesToken) == address(0)) {
+            revert IHyperdriveDeployerCoordinator.InvalidVaultSharesToken();
+        }
+
+        // Ensure that the base token address is properly configured.
+        if (
+            address(_deployConfig.baseToken) !=
+            IAToken(address(_deployConfig.vaultSharesToken))
+                .UNDERLYING_ASSET_ADDRESS()
+        ) {
+            revert IHyperdriveDeployerCoordinator.InvalidBaseToken();
+        }
+
+        // Ensure that the minimum share reserves are large enough to meet the
+        // minimum requirements for safety.
+        //
+        // NOTE: Some pools may require larger minimum share reserves to be
+        // considered safe. This is just a sanity check.
+        if (
+            _deployConfig.minimumShareReserves <
+            10 ** (_deployConfig.baseToken.decimals() - 4)
```

Comment:
This reminds me. I left a TODO in the Operations Playbook on the napkin math you did to determine this value for ETH, DAI, ETC. Can you throw your thoughts down there?  I put a link to the 4626 inflation attack article, but i didn't totally remember you dark magic here,

Reply:
Yeah, I can. I updated this to 1e15 for DAI